### ISSUE TITLE
ARM/M1 Support

### ISSUE THREAD

**Issue by justinrmiller** [dmlc/decord#225]
Hello! We're converting over to M1 Macs at work and I've noticed there doesn't appear to be a decord build form ARMv8/M1. We can emulate x86 in our Docker containers locally but it'd be great if we could run the project natively. Would you be open to adding support for this platform and if so would you like some assistance in doing so?

Thanks,
Justin

**Comment by khurram-nexttier** [comment]
Agreed with @justinrmiller, we need to have support for Apple Silicon. Any help i can provide with that?

**Comment by alkeshpatel11** [comment]
+1. Can we have this much needed support for ARM/M1 ?

**Comment by petered** [comment]
+1 to that - as far as I can tell the only M1-compatible video reader with accurate frame-seeking is pyav, and it's very slow!

**Comment by harrywang** [comment]
+1, any news? Thanks a lot!

**Comment by TanujW** [comment]
+1

**Comment by johnnynunez** [comment]
any news?

**Comment by petered** [comment]
Would any of the decord experts like @zhreshold or @innerlee be able to chime in on what it would take to support M1?

Looks like it would be useful to a lot of people and maybe just having a high-level idea of what needs to be done would encourage someone to take the initiative.

**Comment by suryatejreddy** [comment]
We at [EVA](https://github.com/georgia-tech-db/eva/) have been using decord in our project and find it to be extremely useful! To support eva users working on the newer macs (arm architecture), we have created a fork https://pypi.org/project/eva-decord/ that fixed the installation issues. We would love to add them to decord [@zhreshold](https://github.com/zhreshold), but you can use it for time being. (edited)

**Comment by petered** [comment]
Can confirm that the EVA solution works.  Thank you @suryatejreddy and the EVA people for not only creating a solution but posting it publicly for all to use.

**Comment by mrtngrsbch** [comment]
EVA does not work for me and others.

**Comment by johnnynunez** [comment]
> EVA does not work for me and others.

https://github.com/johnnynunez/decord2
https://pypi.org/project/decord2/1.0.0/

**Comment by djbrickz-cloud** [comment]
> We at [EVA](https://github.com/georgia-tech-db/eva/) have been using decord in our project and find it to be extremely useful! To support eva users working on the newer macs (arm architecture), we have created a fork https://pypi.org/project/eva-decord/ that fixed the installation issues. We would love to add them to decord [@zhreshold](https://github.com/zhreshold), but you can use it for time being. (edited)

I had been in Compile hell for 2 days straight on my M1 (even using GBT) until i ran into this thread. I also threw in the towel but Thanks you so much!!!

**Comment by johnnynunez** [comment]
> > We at [EVA](https://github.com/georgia-tech-db/eva/) have been using decord in our project and find it to be extremely useful! To support eva users working on the newer macs (arm architecture), we have created a fork https://pypi.org/project/eva-decord/ that fixed the installation issues. We would love to add them to decord [@zhreshold](https://github.com/zhreshold), but you can use it for time being. (edited)
> 
> I had been in Compile hell for 2 days straight on my M1 (even using GBT) until i ran into this thread. I also threw in the towel but Thanks you so much!!!

https://pypi.org/project/decord2/
